src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java

```diff
@@ -255,21 +258,35 @@ public String value() {
           + "(NO rate limiting if n=0)";
   private static final int RATE_LIMITING_EVERY_N_DEFAULT = 0;
 
-  public static final String ERRORS_TOLERANCE_CONFIG = "errors.tolerance";
+  public static final String ERRORS_TOLERANCE_CONFIG = "mongo.errors.tolerance";
   public static final String ERRORS_TOLERANCE_DISPLAY = "Error Tolerance";
   public static final ErrorTolerance ERRORS_TOLERANCE_DEFAULT = ErrorTolerance.NONE;
   public static final String ERRORS_TOLERANCE_DOC =
       "Behavior for tolerating errors during connector operation. 'none' is the default value "
           + "and signals that any error will result in an immediate connector task failure; 'all' "
           + "changes the behavior to skip over problematic records.";
 
-  public static final String ERRORS_LOG_ENABLE_CONFIG = "errors.log.enable";
+  @Deprecated
```

Comment:
The `Deprecated` annotation means that it generates warnings whenever the variable is used. However, `LEGACY_ERRORS_TOLERANCE_CONFIG` is "new" and won't be in use anywhere else.

I think that we should leave the `ERRORS_TOLERANCE_CONFIG` with it's current value, and add deprecated there.
Otherwise if someone else uses the `ERRORS_TOLERANCE_CONFIG`, they get force-migrated to the new config value + functionality.

Reply:
Force migration is fine here as it won't alter functionality at all; things will stay exactly the same. Plus, any use of the `ERRORS_TOLERANCE_CONFIG` field is going to be programmatic, and indicates the intent to use the "correct" name for that property; if we're deprecating the old name and introducing a new one, it seems more sensible to make that transition seamless.

Reply:
What i mean is that if someone used `getString(ERRORS_TOLERANCE_CONFIG)`, they'd get a different result before and after this migration. The only reason to add `@Deprecated` is to warn someone using the bare constant that the behavior is being removed.

Reply:
Where will people call `getString(ERRORS_TOLERANCE_CONFIG)` outside of this repo (where all calls have been accounted for)?

I was thinking that marking the constant `@Deprecated` would discourage people from using it at all, without at least being aware that it refers to something that may be removed in a future release. As an alternative, it could be made private if we want to prevent people from programmatically accessing it.

Reply:
> Where will people call getString(ERRORS_TOLERANCE_CONFIG) outside of this repo (where all calls have been accounted for)?

If this change is cherry-picked upstream, or to a different fork, those forks might have bare calls. This fork had bare calls to the config before you migrated them, so it's not crazy to assume that someone might have copy-pasted code (or refactored, or moved a file, etc.

Yes, `@Deprecated` will prevent future use of this variable, but it still allows past use of the variable, which I think is usually the primary use case. It's something that's not covered in the diff right now but could be.